.github/workflows/golangci-lint.yml

```
@@ -0,0 +1,48 @@
+name: golangci-lint
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - main
+  pull_request:
+
+permissions:
+  contents: read
+  # Optional: allow read access to pull request. Use with `only-new-issues` option.
+  # pull-requests: read
+
+jobs:
+  golangci:
```

We have a branch protection setting for the expected checks, and the `static-analysis` is expected by the branch protection.
```suggestion
  static-analysis:
```
If you want to still name it as `golangci` please let me know so we can push our config-as-code for the branch protection